### PR TITLE
username's handled as per RC core repo

### LINKED ISSUE #50: Handle names as per RC core
### Steps to reproduce:

1. Issue that occurred at the [demo page](https://sidmohanty11.github.io/embeddedchat-demo-deploy)

### Expected behavior:
Usernames should have been handled gracefully.

### Actual behavior:
It broke

### PR DESCRIPTION
**Pull request by ayush3160**
# Username's are handled as per the RC core repo

## Acceptance Criteria fulfillment

- [x] Checking If the username already exists or not.
- [x] Support for foreign language name's

Fixes #50

### CODE REVIEW

**Comment by ayush3160** [comment]
Hi @sidmohanty11 , can you please review the PR.

**Comment by sidmohanty11** [comment]
@ayush3160 I think you've missed my comment on this PR. Please take a look into it

**Comment by ayush3160** [comment]
@sidmohanty11 , I am not able to see any comments from your side.

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]
Please convert it to a separate function

File: src/lib/api.js [hunk 1]
```diff
@@ -80,18 +80,33 @@ export default class RocketChatInstance {
...
-        body: `{"userId": "${userid}", "data": { "username": "${newUserName}" }}`,
-        headers: {
-          'Content-Type': 'application/json',
-          'X-Auth-Token': Cookies.get('rc_token'),
-          'X-User-Id': Cookies.get('rc_uid'),
-        },
-        method: 'POST',
-      });
-      return await response.json();
+      const res = await fetch(
```
[the remark this answers is not in the document]
Comment (ayush3160) [reply src/lib/api.js]: @sidmohanty11 , I have separated username suggestion to a different function. Please have a look.

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]
@ayush3160 please take a look.

Also, my suggestion will be to use VSCode's ESlint plugin. Didn't it run husky pre-commit hook when you committed?

File: src/lib/api.js [hunk 2]
```diff
@@ -79,19 +79,42 @@ export default class RocketChatInstance {
...
+        }
+      );
+
+      return await response.json()
+    } catch (error) {
+      console.log(error.message)
+    }
+  }
+
   async updateUserUsername(userid, username) {
```
[the remark this answers is not in the document]
Comment (ayush3160) [reply src/lib/api.js:102]: I have fixed this

File: src/lib/api.js [hunk 3]
```diff
@@ -79,19 +79,42 @@ export default class RocketChatInstance {
...
-        body: `{"userId": "${userid}", "data": { "username": "${newUserName}" }}`,
-        headers: {
-          'Content-Type': 'application/json',
-          'X-Auth-Token': Cookies.get('rc_token'),
-          'X-User-Id': Cookies.get('rc_uid'),
-        },
-        method: 'POST',
-      });
-      return await response.json();
+      const username = await this.getUserNameSuggestion();
```
[the remark this answers is not in the document]
Comment (ayush3160) [reply src/lib/api.js:104]: @sidmohanty11 , Are you suggesting that we should first try to set username with the standard username firstname.lastname and if we get an error from this, Then we should make the username suggestion call and set the username. Let me know If I am getting this right.

[the remark this answers is not in the document]
Comment (sidmohanty11) [reply src/lib/api.js:104]: @ayush3160 yup

[the remark this answers is not in the document]
Comment (ayush3160) [reply src/lib/api.js:104]: Okay, I'll implement that.

[the remark this answers is not in the document]
Comment (ayush3160) [reply src/lib/api.js]: @sidmohanty11 , I have changed the approach please have a look.

[the remark this answers is not in the document]
Comment (sidmohanty11) [reply src/lib/api.js]: Amazing work!

**Review by sidmohanty11 — APPROVED** [review]
LGTM!

**Pull request merged**